File: go-backend/internal/model/conversation_message.go
// Package model 对话消息实体
// @author <a href="https://codefather.cn">编程导航学习圈</a>
package model

import "time"

type ConversationMessage struct {
	ID             string    `gorm:"column:id;type:varchar(36);primaryKey" json:"id"`
	ConversationID string    `gorm:"column:conversationId;type:varchar(36);not null" json:"conversationId"`
	UserID         int64     `gorm:"column:userId;type:bigint;not null" json:"userId"`
	MessageIndex   int       `gorm:"column:messageIndex;type:int;not null" json:"messageIndex"`
	Role           string    `gorm:"column:role;type:varchar(20);not null" json:"role"`
	ModelName      string    `gorm:"column:modelName;type:varchar(100)" json:"modelName"`
	VariantIndex   *int      `gorm:"column:variantIndex;type:int" json:"variantIndex"`
	Content        string    `gorm:"column:content;type:text;not null" json:"content"`
	Images         *string   `gorm:"column:images;type:json" json:"images"`
	ToolsUsed      *string   `gorm:"column:toolsUsed;type:json" json:"toolsUsed"`
	ResponseTimeMs int       `gorm:"column:responseTimeMs;type:int" json:"responseTimeMs"`
	InputTokens    int       `gorm:"column:inputTokens;type:int" json:"inputTokens"`
	OutputTokens   int       `gorm:"column:outputTokens;type:int" json:"outputTokens"`
	Cost           float64   `gorm:"column:cost;type:decimal(10,6)" json:"cost"`
	Reasoning      string    `gorm:"column:reasoning;type:text" json:"reasoning"`
	CodeBlocks     *string   `gorm:"column:codeBlocks;type:text" json:"codeBlocks"`
	CreateTime     time.Time `gorm:"column:createTime;type:datetime" json:"createTime"`
	UpdateTime     time.Time `gorm:"column:updateTime;type:datetime" json:"updateTime"`
	IsDelete       int       `gorm:"column:isDelete;type:tinyint;default:0" json:"isDelete"`
}

func (ConversationMessage) TableName() string {
	return "conversation_message"
}

File: python-backend/app/models/conversation_message.py
"""
对话消息数据库模型
@author <a href="https://codefather.cn">编程导航学习圈</a>
"""
from datetime import datetime
from decimal import Decimal
from sqlalchemy import Column, String, BigInteger, DateTime, Integer, Text, DECIMAL, JSON
from sqlalchemy.orm import declarative_base

Base = declarative_base()


class ConversationMessage(Base):
    """
    对话消息表模型
    """
    __tablename__ = 'conversation_message'

    id = Column(String(36), primary_key=True, comment='消息唯一标识')
    conversation_id = Column('conversationId', String(36), nullable=False, comment='对话ID')
    user_id = Column('userId', BigInteger, nullable=False, comment='用户ID')
    message_index = Column('messageIndex', Integer, nullable=False, comment='消息序号(从0开始)')
    role = Column(String(20), nullable=False, comment='角色: user/assistant')
    model_name = Column('modelName', String(100), nullable=True, comment='模型名称(assistant消息)')
    variant_index = Column('variantIndex', Integer, nullable=True, comment='变体索引(用于prompt_lab，user和assistant消息)')
    content = Column(Text, nullable=False, comment='消息内容')
    images = Column(JSON, nullable=True, comment='图片URL列表')
    tools_used = Column('toolsUsed', JSON, nullable=True, comment='工具调用信息（JSON，含联网搜索关键词/来源等）')
    response_time_ms = Column('responseTimeMs', Integer, nullable=True, comment='响应时间(毫秒)')
    input_tokens = Column('inputTokens', Integer, nullable=True, comment='输入Token数')
    output_tokens = Column('outputTokens', Integer, nullable=True, comment='输出Token数')
    cost = Column(DECIMAL(10, 6), nullable=True, comment='成本(USD)')
    reasoning = Column(Text, nullable=True, comment='思考过程（thinking模式）')
    code_blocks = Column('codeBlocks', Text, nullable=True, comment='代码块列表（JSON格式）')
    create_time = Column('createTime', DateTime, nullable=False, default=datetime.now, comment='创建时间')
    update_time = Column('updateTime', DateTime, nullable=False, default=datetime.now, onupdate=datetime.now, comment='更新时间')
    is_delete = Column('isDelete', Integer, nullable=False, default=0, comment='逻辑删除')

    def __repr__(self):
        return f"<ConversationMessage(id={self.id}, conversation_id={self.conversation_id}, role={self.role})>"
